app.py

```diff
@@ -18,18 +20,19 @@ def root():
 @app.route('/api/definition')
 @cross_origin()
 def definition():
-    word = request.args.get('word', default='', type=str)
-    if word == '':
-        return 'No word was given'
+    letters = request.args.get('letters', default='', type=str)
+    if letters == '':
+        return 'No letters were given'
 
-    results = nearest_neighbor.nearest_neighbors(model, word, vectors)
+    best_word = generator.generate(letters)
+    results = nearest_neighbor.nearest_neighbors(model, best_word, vectors)
     definitions = nearest_neighbor.get_definitions(results)
-    with open('word_generation/definitions.txt', 'w+') as f:
```

Comment:
Now that its all hooked up, I removed the code that was saving the definitions to a file. It can just stay in memory.